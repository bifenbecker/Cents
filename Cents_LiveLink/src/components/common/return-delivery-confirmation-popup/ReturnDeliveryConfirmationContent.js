import React from "react";
import {Box, Flex, Image, Text} from "rebass/styled-components";
import PropTypes from "prop-types";
import {TurnAroundTime} from "../../../assets/images";
import {WE_WILL_TEXT_YOU_WHEN_READY} from "../../../constants/order";
import {businessSettingsSelectors} from "../../../features/business/redux";
import {useAppSelector} from "app/hooks";

const ReturnDeliveryConfirmationContent = ({
  day,
  time,
  turnAroundInHours,
  onScheduleNowClick,
}) => {
  const businessSettings = useAppSelector(
    businessSettingsSelectors.getBusinessSettingsFromRedux
  );
  return (
    <Flex {...styles.mainWrapper}>
      <Image src={TurnAroundTime} {...styles.turnaroundIcon} />
      <Box>
        <Text {...styles.subHeading}>{WE_WILL_TEXT_YOU_WHEN_READY}</Text>
        <Flex {...styles.textWrapper}>
          {businessSettings?.dryCleaningEnabled ? (
            <Text {...styles.normalText}>
              Your laundry should be ready {turnAroundInHours} hours after pickup. We will
              text you when it’s ready so you can schedule your return delivery at your
              convenience.
            </Text>
          ) : (
            <Text {...styles.normalText}>
              Your laundry should be ready by {day} at {time}. We will text you when it’s
              ready so you can schedule your return delivery at your convenience.
            </Text>
          )}
        </Flex>
        <Text {...styles.normalText}>Prefer to preschedule your return delivery?</Text>
        <Flex {...styles.multiTextWrapper}>
          <Text {...styles.multiTextWrapper.normalText}>No problem.</Text>
          <Text {...styles.multiTextWrapper.blueText} onClick={onScheduleNowClick}>
            Click here
          </Text>
        </Flex>
      </Box>
    </Flex>
  );
};

const styles = {
  mainWrapper: {
    sx: {
      flexDirection: "column",
    },
  },
  turnaroundIcon: {
    sx: {
      alignSelf: "center",
      width: "245px",
      height: "200px",
    },
  },
  textWrapper: {
    margin: "20px 0px 20px",
  },
  subHeading: {
    padding: "20px 0px 0px 10px",
  },
  normalText: {
    fontSize: "16px",
    color: "BLACK",
    fontFamily: "secondary",
    sx: {ml: "10px"},
  },

  multiTextWrapper: {
    flexDirection: "row",
    normalText: {
      fontSize: "16px",
      color: "BLACK",
      fontFamily: "secondary",
      sx: {ml: "10px"},
    },
    blueText: {
      color: "primary",
      fontWeight: 500,
      lineHeight: "18px",
      sx: {
        textDecoration: "underline",
        cursor: "pointer",
      },
      pl: "5px",
    },
  },
  gotItButton: {
    sx: {
      backgroundColor: "CENTS_BLUE",
      width: "100%",
      borderRadius: 31,
      textTransform: "uppercase",
    },
    p: "20px",
    mt: "60px",
  },
};

ReturnDeliveryConfirmationContent.propTypes = {
  onScheduleNowClick: PropTypes.func.isRequired,
  day: PropTypes.string.isRequired,
  time: PropTypes.string.isRequired,
};
ReturnDeliveryConfirmationContent.defaultProps = {
  day: "",
  time: "",
};

export default ReturnDeliveryConfirmationContent;
