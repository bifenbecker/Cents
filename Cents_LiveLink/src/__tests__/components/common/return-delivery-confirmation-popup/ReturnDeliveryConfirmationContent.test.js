import React from "react";
import * as redux from "react-redux";
import {render, fireEvent} from "@testing-library/react";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";

import ReturnDeliveryConfirmationContent from "../../../../components/common/return-delivery-confirmation-popup/ReturnDeliveryConfirmationContent";

const day = "12";
const time = "30";
const onScheduleNowClick = jest.fn();

const initialState = {
  businessSettings: {
    data: {
      dryCleaningEnabled: false,
    },
  },
};
const mockStore = configureStore();
let store;

const mockDispatch = jest.fn();
const mockSelector = jest.fn();

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: () => mockDispatch,
  useSelector: () => mockSelector,
}));

const spy = jest.spyOn(redux, "useSelector");
spy.mockReturnValue({
  businessSettings: {
    data: {
      dryCleaningEnabled: false,
    },
  },
});

const renderReturnDeliveryConfirmationContent = () => {
  store = mockStore(initialState);
  const {container, queryByText} = render(
    <Provider store={store}>
      <ReturnDeliveryConfirmationContent
        {...{
          day,
          time,
          onScheduleNowClick,
        }}
      />
    </Provider>
  );

  return {container, queryByText};
};

describe("ReturnDeliveryConfirmationContent Component", () => {
  it("should match snapshot", () => {
    const {container} = renderReturnDeliveryConfirmationContent();
    expect(container).toMatchSnapshot();
  });

  it("should call onScheduleNowClick when `click here` is clicked", () => {
    const {queryByText} = renderReturnDeliveryConfirmationContent();

    const button = queryByText("Click here");
    fireEvent.click(button);

    expect(onScheduleNowClick).toBeCalled();
  });
});
