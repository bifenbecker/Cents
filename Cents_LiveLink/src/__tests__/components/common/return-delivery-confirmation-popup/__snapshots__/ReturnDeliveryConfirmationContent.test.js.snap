// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReturnDeliveryConfirmationContent Component should match snapshot 1`] = `
<div>
  <div
    class="sc-bczRLJ sc-gsnTZi cNLHOo hzPtvw"
  >
    <img
      class="sc-bczRLJ fpnpAl"
      src="TurnaroundTime.svg"
    />
    <div
      class="sc-bczRLJ jITePI"
    >
      <div
        class="sc-bczRLJ ibigaf"
      >
        We’ll text you when it’s ready.
      </div>
      <div
        class="sc-bczRLJ sc-gsnTZi hIKzOH hzPtvw"
      >
        <div
          class="sc-bczRLJ IWDFp"
          color="BLACK"
          font-family="secondary"
          font-size="16px"
        >
          Your laundry should be ready by 
          12
           at 
          30
          . We will text you when it’s ready so you can schedule your return delivery at your convenience.
        </div>
      </div>
      <div
        class="sc-bczRLJ IWDFp"
        color="BLACK"
        font-family="secondary"
        font-size="16px"
      >
        Prefer to preschedule your return delivery?
      </div>
      <div
        class="sc-bczRLJ sc-gsnTZi cdIXbE hzPtvw"
      >
        <div
          class="sc-bczRLJ IWDFp"
          color="BLACK"
          font-family="secondary"
          font-size="16px"
        >
          No problem.
        </div>
        <div
          class="sc-bczRLJ dXFBAw"
          color="primary"
          font-weight="500"
        >
          Click here
        </div>
      </div>
    </div>
  </div>
</div>
`;
